cfpb/sbl-frontend
# [WIP] chore(linting): run a combined eslint, prettier, organize-imports pass

# WIP: Let's wait to review this until after the demo

Previously we've run with automatic fix commands with eslint then prettier. Now that we've combined them and their rules, so we need to run it again to avoid constantly shifting files on save. 

This will hopefully help with the few remaining changes that happen on save, and this is probably a good time to do it before the git conflicts flare up.

Also [VSCode's](https://github.com/cfpb/sbl-frontend/blob/main/.vscode/settings.json#L7) `organize-imports` on save action conflicts with the way ESLint does it, causing it to slightly shift on save even after we run eslint (like in `App.tsx`). Ive disabled that VSCode action and replaced it with `eslint-plugin-simple-import-sort` which allows an easy `--fix` on files.

Also also, disables running prettier by itself in `lint-staged` as part of the git hook, since that causes conflicts with eslint when run by itself.

I've left stylelint for CSS alone for now. I've also not changed the import order of imports of `.css/.less` files, since rarely that import order might matter. I don't really want to change any of those until we set up a visual diff tool in cypress so I don't have to check everything to make sure nothing changed.

This will make PRs easier to parse without a bunch of diff lines unrelated to the PR change. I feel like we're going to want to go through the eslint rule pile when we open up the new repo for the filing app, but this is a good start.

## Changes

- `yarn run-eslint` (which now also runs prettier)

## How to test this PR

1. Go to main on a file with linting errors that would be automatically fixed on saved like `src/api/fetchInstitutions.ts`.
2. Notice that when you save it doesn't perform a bunch of changes.
3. Compare with `main`

## Screenshots

### _Before_

![Screenshot 2024-01-16 at 6 57 27 PM](https://github.com/cfpb/sbl-frontend/assets/19983248/3db98bc7-5453-4d50-b353-50992ccc5b14)


### _After_
![Screenshot 2024-01-16 at 6 56 47 PM](https://github.com/cfpb/sbl-frontend/assets/19983248/66c2ad0b-3d02-42bc-a275-b1c1fb74502a)

## Notes
- If you change an eslint rule in the future (we definitely will), please re-run `yarn run-eslint`
